### PR TITLE
[5.3] optimize validator + benchmarks

### PR DESCRIPTION
**Pull request by petar-dambovaliev**
benchmarks

Long operation is the current code and short operation is my optimization.

Mine is searching in a hash table which is O(n)
the current code uses array_unique which uses O(n·log n)

### CODE REVIEW

**Comment by taylorotwell** [comment]
So those are micro seconds?

**Comment by PetarDambovaliev** [comment]
@taylorotwell Yes

File: src/Illuminate/Validation/Validator.php [hunk 1]
```diff
@@ -601,10 +601,14 @@ protected function attributesThatHaveMessages()
         $results = [];
 
         foreach ($this->messages()->toArray() as $key => $message) {
-            $results[] = explode('.', $key)[0];
+            $key = explode('.', $key)[0];
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: If we're going to optimize this, please pass a 3rd paramater to this function `1`. :)

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by taylorotwell** [comment]
That's such a tiny micro-optimization I think we will just hold off for now.

**Pull request closed without merging**